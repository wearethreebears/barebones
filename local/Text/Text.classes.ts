import type { PropType } from "vue";
import type { ClassProps } from "@jumpstart/types/prop";

export const classProps: ClassProps = {
  style: {
    type: String as PropType<string>,
    default: "default",
  },
  color: {
    type: String as PropType<string>,
    default: "default",
  },
};

export default classProps;
