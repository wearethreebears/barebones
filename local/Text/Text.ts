import type { Parts } from '@jumpstart/types/component';

// Base Classes
const parts: Parts = {
  component: {
    style: {
      'default': ['font-sans'],
    },
    color: {
      'default': ['text-slate-500'],
    }
  }
}

export default parts;
