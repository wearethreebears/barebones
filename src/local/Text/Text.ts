import type { Parts } from "@barebones/types/component";

// Base Classes
const parts: Parts = {
  component: {
    style: {
      default: ["font-sans"],
    },
    color: {
      default: ["text-slate-500"],
    },
  },
};

export default parts;
