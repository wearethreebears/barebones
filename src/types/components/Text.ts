import type { Parts } from "@barebones/types/component";

export interface Text {
  parts: Parts;
}
