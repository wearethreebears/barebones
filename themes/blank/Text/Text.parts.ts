import type { Parts } from "@barebones/types/component";
import type { Text } from "@barebones/types/components/Text";

// Base Classes
export default function TextLocal(): Text {
  const parts: Parts = {
    component: {},
  };

  return {
    parts,
  };
}
