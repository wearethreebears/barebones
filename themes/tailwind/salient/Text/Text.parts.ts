import type { Parts } from "@barebones/types/component";
import type { Text } from "@barebones/types/components/Text";

// Base Classes
export default function TextLocal(): Text {
  const parts: Parts = {
    component: {
      style: {
        default: ["font-sans"],
      },
      color: {
        default: ["text-slate-500"],
      },
    },
  };

  return {
    parts,
  };
}
