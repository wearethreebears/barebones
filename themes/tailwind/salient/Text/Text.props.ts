import type { PropType } from "vue";
import type { ClassProps } from "@barebones/types/prop";

export const classProps: ClassProps = {
  style: {
    type: String as PropType<string>,
    default: "default",
  },
  color: {
    type: String as PropType<string>,
    default: "default",
  },
  size: {
    type: String as PropType<string>,
    default: "default",
  },
};

export default classProps;
