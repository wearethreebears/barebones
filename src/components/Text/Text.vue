<template>
  <component :is="tag" :class="part['component']"><slot /></component>
</template>

<script lang="ts">
export default { name: "BonesText" };
</script>

<script setup lang="ts">
import useComponent from "@barebones/composables/useComponent";

import type { PropType } from "vue";
import type { Text } from "@barebones/types/tag";

import { classProps } from "@barebones-local/Text/Text.classes";
import parts from "@barebones-local/Text/Text";

const props = defineProps({
  ...classProps,
  tag: {
    required: false,
    type: String as PropType<Text>,
    default: "p",
  },
});

const { part } = useComponent(parts, props, classProps);
</script>
