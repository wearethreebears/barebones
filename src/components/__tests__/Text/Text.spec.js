import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";
import { allComponentPartClassesToDisplayCorrectly } from "@barebones/components/__tests__/helpers";
import Text from "@barebones/components/Text/Text.vue";

import TextLocal from "@barebones-local/Text/Text.parts";
const { parts } = TextLocal();


const defaultSlot = "Hello world";

allComponentPartClassesToDisplayCorrectly(Text, parts);

describe("Text", () => {
  let wrapper;
  beforeEach(async () => {
    wrapper = mount(Text, {
      slots: {
        default: defaultSlot
      }
    });
  })

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("displays the correct slot", () => {
    expect(wrapper.html()).contains(defaultSlot)
  });

});
