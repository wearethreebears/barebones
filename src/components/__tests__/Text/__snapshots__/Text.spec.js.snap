// Vitest Snapshot v1

exports[`Text > renders properly 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <p
        class="font-sans text-slate-500"
        data-part="component"
      >
        
        Hello world
        
      </p>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": {
          "_ctx": [Circular],
          "default": [Function],
        },
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "color": "default",
            "style": "default",
            "tag": "p",
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "part": ComputedRefImpl {
              "__v_isReadonly": true,
              "__v_isRef": true,
              "_cacheable": true,
              "_dirty": false,
              "_setter": [Function],
              "_value": {
                "component": [
                  "font-sans",
                  "text-slate-500",
                ],
              },
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Set {
                      [Circular],
                    },
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
              "effect": ReactiveEffect {
                "active": true,
                "computed": [Circular],
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
            },
            "parts": {
              "component": {
                "color": {
                  "default": [
                    "text-slate-500",
                  ],
                },
                "style": {
                  "default": [
                    "font-sans",
                  ],
                },
              },
            },
            "props": {
              "color": "default",
              "style": "default",
              "tag": "p",
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {
            "hasOwnProperty": [Function],
          },
          "exposed": {
            "hasOwnProperty": [Function],
          },
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "color": "default",
            "style": "default",
            "tag": "p",
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "color": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
              "style": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
              "tag": {
                "0": false,
                "1": true,
                "default": "p",
                "required": false,
                "type": [Function],
              },
            },
            [
              "style",
              "color",
              "tag",
            ],
          ],
          "provides": {},
          "proxy": {
            "color": "default",
            "style": "default",
            "tag": "p",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": true,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": {
                    "component": [
                      "font-sans",
                      "text-slate-500",
                    ],
                  },
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": [
                        Set {
                          [Circular],
                        },
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {
              "default": [Function],
            },
          },
          "setupState": {
            "part": {
              "component": [
                "font-sans",
                "text-slate-500",
              ],
            },
            "parts": {
              "component": {
                "color": {
                  "default": [
                    "text-slate-500",
                  ],
                },
                "style": {
                  "default": [
                    "font-sans",
                  ],
                },
              },
            },
            "props": {
              "color": "default",
              "style": "default",
              "tag": "p",
            },
          },
          "slots": {
            "default": [Function],
          },
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": ,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Hello world",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": Hello world,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": ,
                "key": "_default",
                "patchFlag": -2,
                "props": {
                  "key": "_default",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 16,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Fragment),
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": ,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Hello world",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": Hello world,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": ,
                "key": "_default",
                "patchFlag": -2,
                "props": {
                  "key": "_default",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 16,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Fragment),
              },
            ],
            "dynamicProps": [
              "class",
            ],
            "el": <p
              class="font-sans text-slate-500"
              data-part="component"
            >
              
              Hello world
              
            </p>,
            "key": null,
            "patchFlag": 8,
            "props": {
              "class": "font-sans text-slate-500",
              "data-part": "component",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "p",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/liamhall/Documents/web-development/applications/barebones/src/components/Text/Text.vue",
            "__hmrId": "6dc3e16f",
            "components": {},
            "name": "BonesText",
            "props": {
              "color": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
              "style": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
              "tag": {
                "0": false,
                "1": true,
                "default": "p",
                "required": false,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 5,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <p
          class="font-sans text-slate-500"
          data-part="component"
        >
          
          Hello world
          
        </p>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 36,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/Users/liamhall/Documents/web-development/applications/barebones/src/components/Text/Text.vue",
          "__hmrId": "6dc3e16f",
          "components": {},
          "name": "BonesText",
          "props": {
            "color": {
              "0": false,
              "1": true,
              "default": "default",
              "type": [Function],
            },
            "style": {
              "0": false,
              "1": true,
              "default": "default",
              "type": [Function],
            },
            "tag": {
              "0": false,
              "1": true,
              "default": "p",
              "required": false,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 4,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <p
          class="font-sans text-slate-500"
          data-part="component"
        >
          
          Hello world
          
        </p>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 2,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.33",
  },
  "__setProps": [Function],
  "componentVM": {
    "color": "default",
    "style": "default",
    "tag": "p",
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <p
    class="font-sans text-slate-500"
    data-part="component"
  >
    
    Hello world
    
  </p>,
}
`;
